tasks/main.yml

```diff
@@ -475,10 +430,194 @@
         __mssql_conf_setting: "network forceencryption"
         __mssql_conf_setting_value: "{{ '1' if mssql_tls_enable else 'unset' }}"
 
+- name: Configure availability group on the primary node
+  when:
+    - mssql_ha_configure is defined
+    - mssql_ha_configure | bool
+    - mssql_ha_replica_type == 'primary'
+  any_errors_fatal: true
+  block:
+    - name: Ensure the {{ __mssql_server_ha_packages }} package
+      package:
+        name: "{{ __mssql_server_ha_packages }}"
+        state: "{{ 'present' if mssql_ha_configure | bool else 'absent' }}"
+      register: __mssql_server_ha_packages_install
+
+    - name: Enable the hadrenabled setting
+      include_tasks: mssql_conf_setting.yml
+      vars:
+        __mssql_conf_setting: "hadr hadrenabled"
+        __mssql_conf_setting_value: 1
+
+    # flush_handlers task does not support when conditional
+    - name: Restart the mssql-server service if hadrenabled task was changed
+      service:
+        name: mssql-server
+        state: restarted
+      when: (__mssql_conf_set is changed) or
+        (__mssql_server_ha_packages_install is changed)
+
+    - name: Enable AlwaysOn Health events
+      vars:
+        __mssql_input_sql_file: enable_alwayson.j2
+      include_tasks: input_sql_file.yml
+
+    - name: Create master key encryption
+      vars:
+        __mssql_input_sql_file: create_master_key_encryption.j2
+      include_tasks: input_sql_file.yml
+
+    - name: Create and back up DBM certificate
+      vars:
+        __mssql_input_sql_file: create_and_back_up_dbm_cert.j2
+      include_tasks: input_sql_file.yml
+
+    - name: Fetch DBM certificates from the primary node to the control node
+      fetch:
+        src: "{{ item.value}}"
+        dest: "{{ item.key }}"
+        flat: true
+      with_dict:
+        cert: /var/opt/mssql/data/{{ mssql_ha_dbm_cert_name }}.cer
+        key: /var/opt/mssql/data/{{ mssql_ha_dbm_cert_name }}.pvk
+
+    - name: Create database mirroring endpoints
+      vars:
+        __mssql_input_sql_file: create_endpoint.j2
+      include_tasks: input_sql_file.yml
+
+    - name: Get mssql-server version to see if WRITE_LEASE_VALIDITY is available
```

Comment:
The task name doesn't seem to have anything to do with the task action?

Reply:
I've added a comment above to clarify that write_lease_validity is used in configure_ag.j2, there it is added based on the RHEL version.